front/lib/api/assistant/conversation.ts

```diff
@@ -1120,284 +1120,302 @@ export async function* editUserMessage(
     return;
   }
 
-  // In one big transaction creante all Message, UserMessage, AgentMessage and Mention rows.
-  const { userMessage, agentMessages, agentMessageRows } =
-    await front_sequelize.transaction(async (t) => {
-      await getConversationRankVersionLock(conversation, t);
+  // local error class to differentiate from other errors
+  class UserMessageError extends Error {}
```

Comment:
This can definitely go somewhere else, but this type is only used here to check the error type in a typed way. (checking properties on object is not super typescript compliant).

Reply:
Reverted to UserMessageError and added AgentMessageError. 
ConversationError would have been to vague now that we yield an agent_message_error in retry()